packages/core/components/Modal/CodeSnippet/test/ProcessFiles.test.tsx

```diff
@@ -0,0 +1,82 @@
+import { configureMockStore, mergeState } from "@aics/redux-utils";
+import { render, waitFor } from "@testing-library/react";
+import { expect } from "chai";
+import * as React from "react";
+import { Provider } from "react-redux";
+import { createSandbox } from "sinon";
+
+import ProcessFiles from "../ProcessFilesCodeSnippetModal";
+import { initialState, interaction, selection } from "../../../../state";
+
+describe("<ProcessFiles />", () => {
+    const sandbox = createSandbox();
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    it("renders with Setup and Code sections", () => {
+        sandbox.stub(selection.selectors, "getFileSelection").returns({
+            fetchAllDetails: async () => [],
+        } as any);
+
+        const state = mergeState(initialState, {});
+        const { store } = configureMockStore({ state });
+
+        const { getByText } = render(
+            <Provider store={store}>
+                <ProcessFiles onDismiss={() => undefined} />
+            </Provider>
+        );
+
+        expect(getByText("Setup")).to.exist;
+        expect(getByText("Code")).to.exist;
+    });
+
+    it("dispatches a snippet whose setup includes the inferred plugin deps", async () => {
+        sandbox.stub(selection.selectors, "getFileSelection").returns({
+            fetchAllDetails: async () => [
+                { path: "/data/a/sample_a.ome.tiff" }, // -> bioio-ome-tiff
+                { path: "/data/c/sample_c.lif" }, // -> bioio-lif
+                { path: "/data/d/notes.txt" }, // ignored
+            ],
+        } as any);
+
+        const state = mergeState(initialState, {});
+        const { store } = configureMockStore({ state });
+        const origDispatch = store.dispatch;
+        const dispatchSpy = sandbox.spy((action: any) => origDispatch(action));
```

Comment:
per other PR, could just get `actions` from `configureMockStore` and `actions.includesMatch({type, payload})`  below instead of needing to set up a sandbox spy
Similarly, can still use `await logicMiddleware.whenComplete();` instead of needing to `await waitFor`